mail/libraries/lite_to_edifact_converter.py

```
@@ -139,15 +134,21 @@ def licences_to_edifact(licences: QuerySet, run_number: int) -> str:
                     )
             if licence_payload.get("type") in LicenceTypeEnum.OPEN_LICENCES:
                 line_no += 1
-                edifact_file += (
-                    "\n{}\\line\\1\\\\\\\\\\Open Licence goods - see actual licence for information\\".format(line_no)
+                edifact_file += "\n{}\\line\\1\\\\\\\\\\Open Licence goods - see actual licence for information\\".format(
+                    line_no
                 )
         line_no += 1
         edifact_file += "\n{}\\end\\licence\\{}".format(line_no, line_no - start_line)
     line_no += 1
     edifact_file += "\n{}\\fileTrailer\\{}".format(
         line_no, licences.count() + licences.filter(action=LicenceActionEnum.UPDATE).count()
     )
+
+    errors = validate_edifact_file(edifact_file)
+    if errors:
+        logging.error(f"File content not as per specification, {errors}")
```

1. You don't need to return errors from `validate_edifact_file`. You can raise the exception directly from that function. `errors` could be a property on that exception and you even write a `__repr__` function that does something like `'\n'.join(errors)`.

2. You don't need to log the error here. If you just raise it (without catching it), it will appear in the logs.

---

True but my idea is to keep the validator code not coupled because it is at very initial state and testing is easier if they don't have to deal with exceptions, that is why for now it is just returning a list of errors and caller can take the appropriate action giving us some flexibility.